enhancements/machine-api/aws-efa-networking.md

```diff
@@ -0,0 +1,253 @@
+When no value is specified, we will use the `interface` type to maintain backwards compatibility.
+
+```go
+type AWSMachineProviderConfig struct {
+  // Existing fields will not be modified
+  ...
+
+  // NetworkInterfaceType specifies the type of network interface to be used for the primary
+  // network interface.
+  // Valid values are "interface", "efa", and omitted, which means no opinion and the platform
+  // chooses a good default which may change over time.
+  // The current default value is "interface".
+  // Please visit https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html to learn more
+  // about the AWS Elastic Fabric Adapter interface option.
+  // +kubebuilder:validation:Enum:="interface";"efa"
+  // +optional
+  NetworkInterfaceType AWSNetworkInterfaceType `json:"networkInterfaceType,omitempty"`
+}
+
+// AWSNetworkInterfaceType defines the network interface type of the the
+// AWS EC2 network interface.
+type AWSNetworkInterfaceType string
+
+const (
+	// AWSInterfaceNetworkInterfaceType is the default network interface type.
+	// This should be used for standard network operations.
+	AWSInterfaceNetworkInterfaceType AWSNetworkInterfaceType = "interface"
+	// AWSEFANetworkInterfaceType is the Elastic Fabric Adapter network interface type.
+	AWSEFANetworkInterfaceType AWSNetworkInterfaceType = "efa"
+)
+```
+
+### Implementation Details/Notes/Constraints
+
+To enable Machine API to create EFA enabled instances, the above API field will need to be added to the AWS Machine
+ProviderConfig. Once that field is added, if set by a user, we will set the interface to the correct value while
+creating new instances.
+
+```go
+  if machineProviderConfig.NetworkInterfaceType != "" {
+		networkInterfaces[0].InterfaceType = aws.String(string(machineProviderConfig.NetworkInterfaceType))
+	}
+```
+
+It is important to note that OpenShift only supports a single network interface on instances created by Machine API
+today, as such, the EFA network interface will be the primary network interface for the instance.
+
+#### What is EFA?
+
+To quote the [AWS documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html):
+
+> An Elastic Fabric Adapter (EFA) is a network device that you can attach to your Amazon EC2 instance to accelerate
+  High Performance Computing (HPC) and machine learning applications. EFA enables you to achieve the application
+  performance of an on-premises HPC cluster, with the scalability, flexibility, and elasticity provided by the AWS
+  Cloud.
+>
+> EFA provides lower and more consistent latency and higher throughput than the TCP transport traditionally used in
+  cloud-based HPC systems. It enhances the performance of inter-instance communication that is critical for scaling HPC
+  and machine learning applications. It is optimized to work on the existing AWS network infrastructure and it can
+  scale depending on application requirements.
+
+### Risks and Mitigations
```

Comment:
If u want to list more limitations, please look here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa.html